indra/media_plugins/cef/linux/volume_catcher_pipewire.cpp

```diff
@@ -0,0 +1,336 @@
+/** 
+ * @file volume_catcher_pipewire.cpp
+ * @brief A Linux-specific, PipeWire-specific hack to detect and volume-adjust new audio sources
+ *
+ * @cond
+ * $LicenseInfo:firstyear=2010&license=viewerlgpl$
+ * Second Life Viewer Source Code
+ * Copyright (C) 2010, Linden Research, Inc.
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License only.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * 
+ * Linden Research, Inc., 945 Battery Street, San Francisco, CA  94111  USA
+ * $/LicenseInfo$
+ * @endcond
+ */
+
+/*
+  The high-level design is as follows:
+  1) Connect to the PipeWire daemon
+  2) Find all existing and new audio nodes
+  3) Examine PID and parent PID's to see if it belongs to our process
+  4) If so, tell PipeWire to adjust the volume of that node
+  5) Keep a list of all audio nodes and adjust when we setVolume()
+ */
+
+#include "linden_common.h"
+
+#include "volume_catcher_linux.h"
+
+extern "C" {
+#include <spa/pod/builder.h>
+#include <spa/param/props.h>
+}
+
+SymbolGrabber pwSymbolGrabber;
+#undef LL_SYMBOL_GRABBER
+#define LL_SYMBOL_GRABBER pwSymbolGrabber
```

Comment:
Please don't try to use this with macro magic if it can be avoided. The symbol grabber sadly is already bnad enough.

It should be possible to pass the desired grabber into LL_GRAB_SYMBOL.
Or maybe it is an option to rework the symbol grabber in such a way, only one instance is needed rather than two.
Mostly greabSymbols would need some love here, to not return false for the symbols of teh opposite volume catcher
I think with some rework the symbol grabber can be made to load symbols for either implementation.

It should be possible to pass the desired